app/controllers/users_controller.rb

```diff
@@ -9,21 +9,28 @@ def create
     @user = User.new(user_params)
     token = params[:stripeToken]
 
-    if @user.save
-      handle_invitation
-      Stripe.api_key = Rails.configuration.stripe[:secret_key]
+    if @user.valid?
       begin
-        charge = Stripe::Charge.create(
-          :amount => 999, 
-          :currency => "usd",
+        charge = StripeWrapper::Charge.create({
+          :amount => 999,
           :source => token,
           :description => "MyFlix sign up charge for #{@user.email}"
-        )
+        })
       rescue Stripe::CardError => e
```

Comment:
You don't rescue the error here but in StripeWrapper. Also no begin block is necessary.

HOMEWORK: check solutions on right path.